tests/test-sources/modules/performance/combine-plugins.nix

```diff
@@ -6,31 +6,107 @@ let
   pluginCount =
     pkg: files: type:
     builtins.length (builtins.filter (p: p != files) pkg.packpathDirs.myNeovimPackages.${type});
+
+  # Stub plugins
+  mkPlugin =
+    name: args:
+    pkgs.vimUtils.buildVimPlugin (
+      {
+        pname = name;
+        version = "2025-04-27";
+        src = pkgs.runCommand "${name}-source" { } ''
+          mkdir "$out"
+          # Add some colliding files
+          echo "# ${name}" > "$out/README.md"
+          echo "Copyright (c) ${name}" > "$out/LICENSE"
+          mkdir "$out/tests"
+          echo "return '${name}'" > "$out/tests/test.lua"
+          # Add import path
+          mkdir -p "$out/lua/${name}"
+          echo "return '${name}'" > "$out/lua/${name}/init.lua"
+          # Add doc
+          mkdir "$out/doc"
+          echo "*${name}.txt* ${name}" > "$out/doc/${name}.txt"
+        '';
+      }
+      // args
+    );
+  # Simple plugins without any features
+  simplePlugin1 = mkPlugin "simple-plugin-1" { };
+  simplePlugin2 = mkPlugin "simple-plugin-2" { };
+  simplePlugin3 = mkPlugin "simple-plugin-3" { };
+  # Plugins with dependencies
+  pluginWithDeps1 = mkPlugin "plugin-with-deps-1" {
+    dependencies = [ simplePlugin1 ];
+  };
+  pluginWithDeps2 = mkPlugin "plugin-with-deps-2" {
+    dependencies = [ simplePlugin2 ];
+  };
+  pluginWithDeps3 = mkPlugin "plugin-with-deps-3" {
+    dependencies = [ simplePlugin3 ];
+  };
+  # Plugin with recursive dependencies
+  pluginWithRecDeps = mkPlugin "plugin-with-rec-deps" {
+    dependencies = [
+      pluginWithDeps1
+      pluginWithDeps2
+    ];
+  };
+  # Plugin with non-standard files
+  pluginWithExtraFiles = mkPlugin "plugin-with-extra-files" {
+    postInstall = ''
+      mkdir "$out/_extra"
+      touch "$out/_extra/test"
+    '';
+  };
+  # Plugins with Python dependencies
+  pluginWithPyDeps1 = mkPlugin "plugin-with-py-deps-1" {
+    passthru.python3Dependencies = ps: [ ps.pyyaml ];
+  };
+  pluginWithPyDeps2 = mkPlugin "plugin-with-py-deps-2" {
+    passthru.python3Dependencies = ps: [ ps.pyyaml ];
+  };
+  pluginWithPyDeps3 = mkPlugin "plugin-with-py-deps-3" {
+    passthru.python3Dependencies = ps: [ ps.requests ];
+  };
+  # Plugins with Lua dependencies
+  ensureDep =
+    drv: dep:
+    drv.overrideAttrs (prev: {
+      propagatedBuildInputs = lib.unique (
+        prev.propagatedBuildInputs or [ ] ++ [ prev.passthru.lua.pkgs.${dep} ]
+      );
+    });
+  pluginWithLuaDeps1 = ensureDep pkgs.vimPlugins.telescope-nvim "plenary-nvim";
+  pluginWithLuaDeps2 = ensureDep pkgs.vimPlugins.nvim-cmp "plenary-nvim";
+  pluginWithLuaDeps3 = ensureDep pkgs.vimPlugins.gitsigns-nvim "nui-nvim";
 in
 {
   # Test basic functionality
   default =
     { config, ... }:
-    {
+    rec {
```

Comment:
Why do we need `rec` here? Can't we access things recursively via the `config` arg?